sql/parser: remove no longer used keywords

yuzefovich opened the pull request:
**plpgsql/parser: rename bazel target name for the package**

For some reason, Go library bazel target for the PLpgSQL parser package
was named `plpgparser`. This creates a problem with an attempt to add
a test that lives in `parser` (as opposed to `parser_test`), so rename
the target to `parser` which follows the convention elsewhere. (I
eventually decided against adding a test but thought it'd be nice to
clean things up.)

**sql/parser: remove no longer used keywords**

This commit removes a handful of unreserved keywords that are no longer
used. Additionally, it refactors existing `reserved_keywords.awk` into
a shell script that is run via a unit test to assert that no longer used
keywords get removed promptly. (I tried applying the same script to
PLpgSQL and JSONPath parsers, and it found nothing for the latter and
produced many false positives for the former, so I decided to not
include the corresponding test.)

Epic: None
Release note: None

yuzefovich committed:
plpgsql/parser: rename bazel target name for the package

For some reason, Go library bazel target for the PLpgSQL parser package
was named `plpgparser`. This creates a problem with an attempt to add
a test that lives in `parser` (as opposed to `parser_test`), so rename
the target to `parser` which follows the convention elsewhere. (I
eventually decided against adding a test but thought it'd be nice to
clean things up.)

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/147386)
<!-- Reviewable:end -->

yuzefovich committed:
sql/parser: remove no longer used keywords

This commit removes a handful of unreserved keywords that are no longer
used. Additionally, it refactors existing `reserved_keywords.awk` into
a shell script that is run via a unit test to assert that no longer used
keywords get removed promptly. (I tried applying the same script to
PLpgSQL and JSONPath parsers, and it found nothing for the latter and
produced many false positives for the former, so I decided to not
include the corresponding test.)

Release note: None

DrewKimball reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 10 of 10 files at r1, 6 of 6 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/147386#-:-ORRnLVi2mJ1sn-ZzcZz:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @yuzefovich)



<!-- Sent from Reviewable.io -->
